qualtran/bloqs/arithmetic/sorting.py

```diff
@@ -48,82 +64,342 @@ class Comparator(Bloq):
 
     @property
     def signature(self):
+        input_dtype = QUInt(self.bitsize)
         return Signature(
             [
-                Register('a', QAny(self.bitsize)),
-                Register('b', QAny(self.bitsize)),
+                Register('a', input_dtype),
+                Register('b', input_dtype),
                 Register('out', QBit(), side=Side.RIGHT),
             ]
         )
 
     def pretty_name(self) -> str:
         return "Cmprtr"
 
-    def _t_complexity_(self):
-        # complexity is from less than on two n qubit numbers + controlled swap
-        # Hard code for now until CSwap-Bloq is merged.
-        # See: https://github.com/quantumlib/Qualtran/issues/219
-        t_complexity = GreaterThan(self.bitsize, self.bitsize).t_complexity()
-        t_complexity += TComplexity(t=14 * self.bitsize)
-        return t_complexity
+    def build_composite_bloq(
+        self, bb: 'BloqBuilder', a: 'Soquet', b: 'Soquet'
+    ) -> dict[str, 'SoquetT']:
+        out = bb.allocate(dtype=QBit())
+        a, b, out = bb.add(GreaterThan(self.bitsize, self.bitsize), a=a, b=b, target=out)
+        out, a, b = bb.add(CSwap(self.bitsize), ctrl=out, x=a, y=b)
+        return {'a': a, 'b': b, 'out': out}
 
 
 @bloq_example
-def _cmp_symb() -> Comparator:
+def _comparator() -> Comparator:
+    comparator = Comparator(7)
+    return comparator
+
+
+@bloq_example
+def _comparator_symb() -> Comparator:
     n = sympy.Symbol('n')
-    cmp_symb = Comparator(bitsize=n)
-    return cmp_symb
+    comparator_symb = Comparator(n)
+    return comparator_symb
+
+
+_COMPARATOR_DOC = BloqDocSpec(bloq_cls=Comparator, examples=[_comparator, _comparator_symb])
+
+
+@frozen
+class ParallelComparators(Bloq):
+    r"""Given k n-bit numbers, for each pair that is `offset` apart, compare and swap if needed to order them.
+
+    For each block of `2 * offset` numbers, apply a `Comparator` between each pair that is `offset` apart.
+    For an offset of $\delta$, this requires
+
+    $$
+        \delta \left\lfloor \frac{k}{2\delta} \right\rfloor
+        + \max((k\mod 2\delta) - \delta, 0)
+    $$
+
+    The above expression is at most $k / 2$.
```

Comment:
There seems to be a break after "this requires, <math expression> . The above expression".

We should finish the "this requires," line. Maybe something like
```suggestion
    number of comparisons. The above expression is at most $k / 2$.
```

Reply:
oops, fixed